src/main/java/com/slisenko/education/html/PageDownloader.java

```diff
@@ -0,0 +1,48 @@
+package com.slisenko.education.html;
+
+import java.io.*;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+/**
+ * Class for downloading resources from the internet
+ */
+public class PageDownloader {
+
+    private final int SIZE_BUF = 256;
+
+    /**
+     * Downloads a resource from the internet and saves it in a file
+     * @param urlStr Link to the resource
+     * @param pathStr File path
+     * @return Returns true if successful, otherwise returns false
+     */
+    public boolean download(String urlStr, String pathStr) {
+
+        URL url;
+
+        try {
+            url = new URL(urlStr);
+        } catch (MalformedURLException e) {
+            System.err.println(this.getClass().getSimpleName() + ":MalformedURLException:" + e.getMessage());
+            return false;
+        }
+
+        try (InputStream reader = new BufferedInputStream(url.openStream());
+             OutputStream writer = new BufferedOutputStream(new FileOutputStream(pathStr));) {
+
+            int sizeReceived;
+            byte[] buf = new byte[SIZE_BUF];
+
+            while ((sizeReceived = reader.read(buf)) > 0) {
```

Comment:
I would use BufferedReader and FileWriter
buffered reader has method readLine, so you can read line by line